relays/ethereum/src/substrate_client.rs

```diff
@@ -281,14 +293,50 @@ impl SubmitEthereumHeaders for SubstrateRpcClient {
 	}
 }
 
+/// A trait for RPC calls which are used to submit proof of Ethereum exchange transaction to a
+/// Substrate runtime. These are typically calls which use a combination of other low-level RPC
+/// calls.
+#[async_trait]
+pub trait SubmitEthereumExchangeTransactionProof: SubstrateRpc {
+	/// Submits Ethereum exchange transaction proof to Substrate runtime.
+	async fn submit_exchange_transaction_proof(
+		&self,
+		params: SubstrateSigningParams,
+		proof: bridge_node_runtime::exchange::EthereumTransactionInclusionProof,
+	) -> Result<()>;
+}
+
+#[async_trait]
+impl SubmitEthereumExchangeTransactionProof for SubstrateRpcClient {
+	async fn submit_exchange_transaction_proof(
+		&self,
+		params: SubstrateSigningParams,
+		proof: bridge_node_runtime::exchange::EthereumTransactionInclusionProof,
+	) -> Result<()> {
+		let account_id = params.signer.public().as_array_ref().clone().into();
+		let nonce = self.next_account_index(account_id).await?;
+
+		let transaction = create_signed_transaction(
+			bridge_node_runtime::Call::BridgeCurrencyExchange(
+				bridge_node_runtime::BridgeCurrencyExchangeCall::import_peer_transaction(proof),
+			),
```

Comment:
I'm a little confused at the wrapped `Call`s here. So the inner call is to ask the Substrate node to import a "lock funds" transaction, what's the outer call for?

Reply:
There are two `Call` enums (as well as `Event`, `Digest`, ...) on different levels. First one is at pallet level - it just enumerates all calls that pallet provides. The second one is on runtime level - it includes calls from all runtime' pallets. So something like that:
```
enum BridgeCall { ImportHeader };
enum ExchangeCall { ImportTransaction };
enum RuntimeCall {
    Bridge(BridgeCall),
    Exchange(ExchangeCall),
}
```